packages/frontend/src/pages/ViewSqlChart.tsx

```diff
@@ -110,6 +117,11 @@ const ViewSqlChart = () => {
         limit: sqlChart?.limit,
     });
 
+    useEffect(() => {
+        if (!chartVizQuery?.data?.url) return;
+        dispatch(setDataUrl(chartVizQuery.data.url));
```

Comment:
I don't really use the dataUrl anymore, because I have direct acces sto chartVizQuery.data in this component , but I think it might be useful for downloading from other places. But let me know if you want me to remove it.

Reply:
that's great! As long as it's managed in state somehow that's good 💪